l10n_it_riba_sale_commission/models/account_move.py

```diff
@@ -0,0 +1,64 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date, timedelta
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    no_commission = fields.Boolean(string="Without commissions")
+
+    def _get_reconciled_invoices_partials(self):
+        """
+        If a partial payment is an sbf Riba payment and the safety days haven't
+        passed yet, I exclude the payment from the partials list where commissions
+        will be generated.
+        """
+        (
+            invoice_partials,
+            exchange_diff_moves,
+        ) = super()._get_reconciled_invoices_partials()
+        to_remove = []
+        for (
+            partial,
+            amount,
+            counterpart_line,
+        ) in invoice_partials:
+            riba_mv_line = self.env["riba.slip.move.line"].search(
+                [("move_line_id", "=", partial.debit_move_id.id)]
+            )
+            if riba_mv_line:
+                riba_type = riba_mv_line.riba_line_id.type
+                if (
+                    riba_mv_line.riba_line_id.due_date
+                    + timedelta(days=riba_mv_line.riba_line_id.config_id.safety_days)
+                    > date.today()
+                ) and riba_type == "sbf":
+                    to_remove.append((partial, amount, counterpart_line))
+        return [
+            item for item in invoice_partials if item not in to_remove
+        ], exchange_diff_moves
+
+
+class AccountInvoiceLineAgent(models.Model):
+    _inherit = "account.invoice.line.agent"
+
+    def _skip_future_payments(self, date_payment_to):
+        if self.invoice_id.is_riba_payment:
+            dates = [
+                line.date_maturity
+                + timedelta(
+                    max(
+                        line.mapped(
+                            "distinta_line_ids.riba_line_id.config_id.safety_days"
+                        ),
+                        default=0,
+                    )
+                )
+                for line in self.invoice_id.line_ids.filtered(lambda r: r.date_maturity)
+            ]
+            if dates:
+                return date_payment_to < max(dates)
+        return super()._skip_future_payments(date_payment_to)
```

Comment:
Non trovo questo metodo in `super`, puoi controllare?
Ho fatto qualche ricerca ed era stato introdotto in `14.0` da https://github.com/OCA/commission/pull/441 ma in `16.0` non lo vedo, e non vedo nemmeno delle PR per portare https://github.com/OCA/commission/pull/441 in `16.0`.

Reply:
> Grazie della review, si, effettivamente il metodo non è presente nella 16. Mi consigli di proporre una pr per portare la pr in account_commission oppure rimuovo il metodo?

Da quello che sto capendo di questi moduli questo override è un punto cardine del funzionamento di questo modulo, e se non esiste in `super` probabilmente vuole anche dire che non viene chiamato quando dovrebbe: secondo me sarebbe da portare https://github.com/OCA/commission/pull/441 a `16.0`.

Mi stupisce anche un po' che i test passino senza che questo metodo sia in `super`, quindi magari mi sto perdendo qualche pezzo.

Reply:
Ho proposto questa pr: https://github.com/OCA/commission/pull/588